CASMTRIAGE-7226 Docs: cfs-api pods in CLBO due to absence of Kafka pods (#5516)
Cray-HPE/docs-csm: troubleshooting/known_issues/cfs-api_pods_in_CLBO_state.md

Reviewer: Please add an entry to the troubleshooting table of contents under [Known issues](https://github.com/Cray-HPE/docs-csm/tree/6ce5d270c645f5bc5d61214b9c7f1f532778fceb/troubleshooting#known-issues).

Author: Done

# `CFS-API` pods in CLBO state during CSM install

## Issue Description

> When installing CSM 1.6, `cray-shared-kafka-kafka-*` pods in services namespace fail to come up which would result in `CFS-API` pods in CLBO state. This happens because of an issue with Zookeeper related to slow DNS.

Reviewer 2: ```suggestion
When installing CSM 1.6, `cray-shared-kafka-kafka-*` pods in the services namespace fail to come up which results in `CFS-API` pods in CLBO state. This happens because of an issue with Zookeeper related to slow DNS.
```

Zookeeper fails to come up if the DNS is not set up for all hosts at startup. When this happens, the cluster gets stuck at Zookeeper pods running, but brokers not coming up.

### Related Issue

- [Zookeeper Issue #4708](https://issues.apache.org/jira/browse/ZOOKEEPER-4708)

## Error Identification

> When the issue occurs the `cray-shared-kafka-kafka-*` pods in services namespace fail to come up and would not be present.

Reviewer 2: ```suggestion
When the issue occurs, the `cray-shared-kafka-kafka-*` pods in the services namespace fail to come up and will not be present.
```

> Also, `CFS-API` pods would be in CLBO state.

Reviewer 2: ```suggestion
Also, `CFS-API` pods will be in CLBO state.
```

> The logs from `strimzi-cluster-operator-*` pod in operators namespace would be throwing errors as follows -

Reviewer 2: ```suggestion
The logs from `strimzi-cluster-operator-*` pod in the operators namespace will be throwing errors as follows:
```

> ```sh

Reviewer: ```suggestion
```text
```

2024-10-04T22:21:54.061644246Z 2024-10-04 22:21:54 WARN  ZookeeperScaler:157 - Reconciliation #1(watch) Kafka(services/cray-shared-kafka): Failed to connect to Zookeeper cray-shared-kafka-zookeeper-0.cray-shared-kafka-zookeeper-nodes.services.svc:2181,cray-shared-kafka-zookeeper-1.cray-shared-kafka-zookeeper-nodes.services.svc:2181,cray-shared-kafka-zookeeper-2.cray-shared-kafka-zookeeper-nodes.services.svc:2181. Connection was not ready in 300000 ms.
2024-10-04T22:21:54.466771715Z 2024-10-04 22:21:54 WARN  ZooKeeperReconciler:834 - Reconciliation #1(watch) Kafka(services/cray-shared-kafka): Failed to verify Zookeeper configuration
```

## Error Conditions

This problem can be triggered by events like:

- Slow DNS propagation to Kubernetes DNS subsystem

## Fix Description

The workaround is to delete the zookeeper pods and let them be re-created by the Strimzi operator.

```bash

> kubectl delete pods -n services `cray-shared-kafka-zookeeper-*`

Reviewer 3: Back ticks in shell have special meaning. If they are here to avoid spell checker errors, need to find another way of spell checker tuning. back ticks will cause shell error if command is copy-pasted into terminal.

Author: Have updated the command to use selector instead
